tests/e2e/monovertex.go

```diff
@@ -362,8 +362,11 @@ func UpdateMonoVertexRollout(name string, newSpec numaflowv1.MonoVertexSpec, exp
 		rollout.Spec.MonoVertex.Spec.Raw = rawSpec
 		return rollout, nil
 	})
+	Document("Verifying MonoVertex spec got updated")
+	// get Pipeline to check that spec has been updated to correct spec
+	VerifyMonoVertexSpec(Namespace, name, verifySpecFunc)
```

Comment:
this fixes the unit test - before we weren't waiting for the spec to get updated, which means that we could start verifying the previous "promoted" MonoVertex for various checks, rather than verifying the new one we're updating